plugin/loader/initializer.go

```diff
@@ -20,24 +20,37 @@ func initialize(plugins []plugin.Plugin) error {
 
 func run(plugins []plugin.Plugin) error {
 	for _, pl := range plugins {
-		err := runIPLDPlugin(pl)
-		if err != nil {
-			return err
+		switch pl.(type) {
+		case plugin.PluginIPLD:
+			err := runIPLDPlugin(pl.(plugin.PluginIPLD))
```

Comment:
no need to do the cast here, the type switch does some magic for you.

Reply:
Are we sure about this? building with `err := runIPLDPlugin(pl)` instead of `err := runIPLDPlugin(pl.(plugin.PluginIPLD))` fails

Reply:
```
plugin/loader/initializer.go:26:24: cannot use pl (type "github.com/ipfs/go-ipfs/plugin".Plugin) as type "github.com/ipfs/go-ipfs/plugin".PluginIPLD in argument to runIPLDPlugin:
        "github.com/ipfs/go-ipfs/plugin".Plugin does not implement "github.com/ipfs/go-ipfs/plugin".PluginIPLD (missing RegisterBlockDecoders method)
```

Reply:
Ah, i forgot one bit, it needs to be:
```go
switch pl := pl.(type) {
```